thrust/iterator/transform_output_iterator.h

```diff
@@ -61,78 +61,78 @@ THRUST_NAMESPACE_BEGIN
  *      return sqrtf(x);
  *    }
  *  };
- *  
+ *
  *  int main()
  *  {
  *    thrust::device_vector<float> v(4);
  *
  *    typedef thrust::device_vector<float>::iterator FloatIterator;
  *    thrust::transform_output_iterator<square_root, FloatIterator> iter(v.begin(), square_root());
  *
- *    iter[0] =  1.0f;    // stores sqrtf( 1.0f) 
+ *    iter[0] =  1.0f;    // stores sqrtf( 1.0f)
  *    iter[1] =  4.0f;    // stores sqrtf( 4.0f)
  *    iter[2] =  9.0f;    // stores sqrtf( 9.0f)
  *    iter[3] = 16.0f;    // stores sqrtf(16.0f)
  *    // iter[4] is an out-of-bounds error
- *                                                                                           
+ *
  *    v[0]; // returns 1.0f;
  *    v[1]; // returns 2.0f;
  *    v[2]; // returns 3.0f;
  *    v[3]; // returns 4.0f;
- *                                                                                           
+ *
  *  }
  *  \endcode
  *
  *  \see make_transform_output_iterator
  */
 
 template <typename UnaryFunction, typename OutputIterator>
-  class transform_output_iterator
+class transform_output_iterator
     : public detail::transform_output_iterator_base<UnaryFunction, OutputIterator>::type
 {
 
   /*! \cond
    */
 
-  public:
-
-    typedef typename
-    detail::transform_output_iterator_base<UnaryFunction, OutputIterator>::type
-    super_t;
+public:
+  typedef
+    typename detail::transform_output_iterator_base<UnaryFunction, OutputIterator>::type super_t;
 
-    friend class thrust::iterator_core_access;
+  friend class thrust::iterator_core_access;
   /*! \endcond
    */
 
+  /*! Null constructor does nothing.
+   */
+  __host__ __device__ transform_output_iterator() {}
```

Comment:
Ditto default rather than define
```suggestion
  transform_output_iterator() = default;
```